shared/domains/chelonia/chelonia.js

```diff
@@ -1177,6 +1208,110 @@ function contractNameFromAction (action: string): string {
   return contractName
 }
 
+function generateEncryptedOrUnencryptedMessage ({
+  contractID,
+  contractName,
+  innerSigningKeyId,
+  encryptionKeyId,
+  signingKeyId,
+  data
+}: {
+  contractID: string,
+  contractName: string,
+  innerSigningKeyId: ?string,
+  encryptionKeyId: ?string,
+  signingKeyId: string,
+  data: Object
+}) {
+  const instance = this._instance
+  const manifestHash = this.config.contracts.manifests[contractName]
+  const { contract } = this.manifestToContract[manifestHash]
+  const state = contract.state(contractID)
+  const signedMessage = innerSigningKeyId
+    ? (state._vm.authorizedKeys[innerSigningKeyId] && state._vm.authorizedKeys[innerSigningKeyId]?._notAfterHeight == null)
+        ? signedOutgoingData(contractID, innerSigningKeyId, (data: any), this.transientSecretKeys)
+        : signedOutgoingDataWithRawKey(this.transientSecretKeys[innerSigningKeyId], (data: any), this.transientSecretKeys)
+    : data
+  const payload = !encryptionKeyId
+    ? signedMessage
+    : encryptedOutgoingData(contractID, encryptionKeyId, signedMessage)
+  const message = signedOutgoingData(contractID, signingKeyId, (payload: any), this.transientSecretKeys)
+  const self = this
+  return {
+    get serialize () {
+      return () => {
+        if (instance !== self._instance) throw new Error('Chelonia instance has been reset')
+        const rootState = sbp(self.config.stateSelector)
+        const height = String(rootState.contracts[contractID].height)
+        return { ...message.serialize(height), height }
+      }
+    }
+  }
```

Comment:
If `.serialize()` is called immediately, and we don't need this outer wrapper object, can we just immediately return line 1246?